lib/new_impl/class_relationships.rb

```diff
@@ -0,0 +1,69 @@
+module NewImpl
```

Comment:
At the risk of causing confusion (gulp!), note that we could promote the `Toy` prefix in `ToyObject`, `ToyModule` and `ToyClass` into a namespace by naming this module `Toy` instead of `NewImpl`, giving us `Toy::Object`, `Toy::Module` and `Toy::Class`.

Reply:
Woah, very cool! TBH I mostly intended the namespacing module to be a standin while we work on integrating our original classes into this new approach, so I was envisioning this going away soon. I'll address your comment below and fully qualify them though!